### ISSUE TITLE
Crypt puts moov/trak/senc box into init.mp4 file, it breaks one of the major hbbtv televisions

### ISSUE THREAD

**Issue by Murmur** [gpac/gpac#932]
Running an encryption mp4box inserts `moov/trak/senc` box inside the init.mp4 segments(video,audio). This is not the problem 99% of the time but one of the major hbbtv brands is unable to playback drm protected video created by mp4box.  

If I delete the senc(Sample Encryption Box) from all the init segments then stream is playable. All the other 99% of devices are also able to playback it as well. Also so is MSEdge(win10,playready), Firefox(win10,widevine+clearkey), Chrome(win10,widevine+clearkey), Operabeta(win10,widevine+clearkey) both streams are fine.

I was trying to study specifications whether it's against the specs to put senc inside the moov box. I tend to think it should go inside the `moof/traf` box but then the specs are not always very clear. Anyone has an opinion on this issue is this a bug in mp4box crypt segmenter?

```
MP4Box.exe -crypt gpacdrm.xml temp-v1.mp4 -out drm/temp-v1.mp4
MP4Box.exe -crypt gpacdrm.xml temp-v2.mp4 -out drm/temp-v2.mp4
MP4Box.exe -crypt gpacdrm.xml temp-v3.mp4 -out drm/temp-v3.mp4
MP4Box.exe -crypt gpacdrm.xml temp-a1.mp4 -out drm/temp-a1.mp4
MP4Box.exe -dash 6000 -frag 6000 -mem-frags -rap -profile dashavc264:live -profile-ext urn:hbbtv:dash:profile:isoff-live:2012 -min-buffer 3000 -mpd-title refapp -mpd-info-url http://refapp -bs-switching no -sample-groups-traf -single-traf -subsegs-per-sidx 1 -segment-name $RepresentationID$_$Number$$Init=i$ -segment-timeline -out manifest.mpd temp-v1.mp4#trackID=1:id=v1:period=p0 temp-v2.mp4#trackID=1:id=v2:period=p0 temp-v3.mp4#trackID=1:id=v3:period=p0 temp-a1.mp4#trackID=1:id=a1:period=p0
```

**gpacdrm.xml**
<?xml version="1.0" encoding="UTF-8" ?>
```
<GPACDRM type="CENC AES-CTR">
<!-- 
  kid=0x43215678123412341234123412341236
  key=0x12341234123412341234123412341236
  iv=0x22ee7d4745d3a26a
--> 

<!-- Playready -->
<DRMInfo type="pssh" version="0">
  <BS ID128="9A04F07998404286AB92E65BE0885F95"/>
  <BS bits="32" endian="little" value="518"/>
  <BS bits="16" endian="little" value="1"/>
  <BS bits="16" endian="little" value="1"/>
  <BS bits="16" endian="little" value="508"/>
  <BS data64="PABXAFIATQBIAEUAQQBEAEUAUgAgAHgAbQBsAG4AcwA9ACIAaAB0AHQAcAA6AC8ALwBzAGMAaABlAG0AYQBzAC4AbQBpAGMAcgBvAHMAbwBmAHQALgBjAG8AbQAvAEQAUgBNAC8AMgAwADAANwAvADAAMwAvAFAAbABhAHkAUgBlAGEAZAB5AEgAZQBhAGQAZQByACIAIAB2AGUAcgBzAGkAbwBuAD0AIgA0AC4AMAAuADAALgAwACIAPgA8AEQAQQBUAEEAPgA8AFAAUgBPAFQARQBDAFQASQBOAEYATwA+ADwASwBFAFkATABFAE4APgAxADYAPAAvAEsARQBZAEwARQBOAD4APABBAEwARwBJAEQAPgBBAEUAUwBDAFQAUgA8AC8AQQBMAEcASQBEAD4APAAvAFAAUgBPAFQARQBDAFQASQBOAEYATwA+ADwASwBJAEQAPgBlAEYAWQBoAFEAegBRAFMATgBCAEkAUwBOAEIASQAwAEUAagBRAFMATgBnAD0APQA8AC8ASwBJAEQAPgA8AEMASABFAEMASwBTAFUATQA+ACsAOQBrAGkANwAvAHQAdABzAGUAQQA9ADwALwBDAEgARQBDAEsAUwBVAE0APgA8AC8ARABBAFQAQQA+ADwALwBXAFIATQBIAEUAQQBEAEUAUgA+AA=="/>
</DRMInfo>

<!-- Widevine -->
<DRMInfo type="pssh" version="0">
  <BS ID128="EDEF8BA979D64ACEA3C827DCD51D21ED"/>
  <BS data="0x08011210"/>
  <BS ID128="43215678123412341234123412341236"/>
</DRMInfo>

<!-- Marlin -->
<DRMInfo type="pssh" version="0">
  <BS ID128="69f908af481646ea910ccd5dcccb0a3a"/>
  <BS data="0x000000186d61726c000000106d6b69640000000000000000"/>
</DRMInfo>

<!-- CENC -->
<DRMInfo type="pssh" version="1">
  <BS ID128="1077efecc0b24d02ace33c1e52e2fb4b"/>
  <BS bits="32" value="1"/>
  <BS ID128="43215678123412341234123412341236"/>
</DRMInfo>

<CrypTrack trackID="1" IsEncrypted="1" IV_size="8" first_IV="0x22ee7d4745d3a26a" saiSavedBox="senc">
  <key KID="0x43215678123412341234123412341236" value="0x12341234123412341234123412341236"/>
</CrypTrack>

</GPACDRM>
```

**Comment by srampur** [comment]
'senc' atom to be in the 'traf' atom.  Having this in any other atom created an issue (like Macroblocks) for me .
Refer https://uvcentral.com/sites/default/files/files/PublicSpecs/CFFMediaFormat-2_1.pdf

**Comment by jeanlf** [comment]
this should be fixed in GPAC 0.9.0
